src/terminal_renderer/TextRenderer.cpp

```diff
@@ -550,7 +550,8 @@ void ComplexTextShaper::endSequence()
 text::shape_result const& ComplexTextShaper::cachedGlyphPositions()
 {
     auto const codepoints = u32string_view(codepoints_.data(), codepoints_.size());
-    if (auto const cached = cache_.find(TextCacheKey{codepoints, style_}); cached != cache_.end())
+    auto const cached = cache_.find(TextCacheKey{codepoints, style_});
+    if (cached != cache_.end())
```

Comment:
I think this is logic error, unless this is some C++ thing I don't understand.

Reply:
Not sure what exactly you meant but that's what it is doing:

check if `TextCacheKey{codepoints, style_}` is in cache, and if so, return that.
otherwise, create cache entry and return that.

The syntax i used is:

```cpp
if (initializer; expression)
    body;
```
just like for-loop initializer, you have that now (C++11) also on if-expression.